### PR TITLE
Prevent underflow with CPU time counter

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

**Comment by lucab** [comment]
Thanks for the PR! I've left a minor comment in review.

Did you hit this in the wild? How did you end up having multiple `ProcessCollector` inside the same application?

**Comment by nox** [comment]
> Thanks for the PR! I've left a minor comment in review.

You didn't.

> How did you end up having multiple `ProcessCollector` inside the same application?

You don't need multiple process collectors, only multiple threads collecting the global registry at the same time.

File: src/process_collector.rs [hunk 1]
```diff
@@ -165,7 +165,12 @@ impl Collector for ProcessCollector {
             // cpu
             let total = (stat.utime + stat.stime) / *CLK_TCK as u64;
             let past = self.cpu_total.get();
-            self.cpu_total.inc_by(total - past);
+            // If two threads are collecting metrics at the same time,
+            // the cpu_total counter may have already been updated,
+            // and the subtraction may underflow.
+            if total < past {
```
[the remark this answers is not in the document]
Comment (nox) [reply src/process_collector.rs]: Done

[the remark this answers is not in the document]
Comment (nox) [reply src/process_collector.rs]: Changed it to using `saturating_sub` instead.

**Review by lucab — APPROVED** [review]
LGTM, thanks!

**Pull request merged**